lib/ogr/spatial_reference.rb

```diff
@@ -2,71 +2,86 @@
 
 require_relative '../ogr'
 require_relative '../gdal'
-require_relative 'spatial_reference_mixins/coordinate_system_getter_setters'
-require_relative 'spatial_reference_mixins/exporters'
-require_relative 'spatial_reference_mixins/importers'
-require_relative 'spatial_reference_mixins/morphers'
-require_relative 'spatial_reference_mixins/parameter_getter_setters'
-require_relative 'spatial_reference_mixins/type_checks'
 require_relative 'new_borrowed'
 
 module OGR
   # Represents a geographic coordinate system.  There are two primary types:
   #   1. "geographic", where positions are measured in long/lat.
   #   2. "projected", where positions are measure in meters or feet.
   class SpatialReference
-    extend OGR::NewBorrowed
-    include GDAL::Logger
-    include SpatialReferenceMixins::CoordinateSystemGetterSetters
-    include SpatialReferenceMixins::Exporters
-    include SpatialReferenceMixins::Importers
-    include SpatialReferenceMixins::Morphers
-    include SpatialReferenceMixins::ParameterGetterSetters
-    include SpatialReferenceMixins::TypeChecks
-
-    # class_eval FFI::OGR::SRSAPI::SRS_UL.to_ruby
-    FFI::OGR::SRSAPI::SRS_UL.constants.each do |_name, obj|
-      const_set(obj.ruby_name, obj.value)
+    autoload :CoordinateSystemGetterSetters,
+             File.expand_path('spatial_reference/coordinate_system_getter_setters', __dir__ || '')
+    autoload :Exporters,
+             File.expand_path('spatial_reference/exporters', __dir__ || '')
+    autoload :Importers,
+             File.expand_path('spatial_reference/importers', __dir__ || '')
+    autoload :Morphers,
+             File.expand_path('spatial_reference/morphers', __dir__ || '')
+    autoload :ParameterGetterSetters,
+             File.expand_path('spatial_reference/parameter_getter_setters', __dir__ || '')
+    autoload :TypeChecks,
+             File.expand_path('spatial_reference/type_checks', __dir__ || '')
+
+    class AutoPointer < ::FFI::AutoPointer
+      # @param pointer [FFI::Pointer]
+      def self.release(pointer)
+        return unless pointer && !pointer.null?
+
+        FFI::OGR::SRSAPI.OSRRelease(pointer)
+      end
     end
 
-    METER_TO_METER = 1.0
+    # @param srs_wkt [String, nil] An optional string argument which if passed
+    #   should be a WKT representation of an SRS. Passing this is equivalent to
+    #   not passing it, and then calling #import_from_wkt with the WKT string.
+    # @return [OGR::SpatialReference]
+    def self.create(srs_wkt = nil)
```

Comment:
This change will probably cause the biggest amount of churn in agrian things, I'd think.